src/sync/sync_user.hpp

```diff
@@ -31,6 +31,16 @@ namespace realm {
 
 class SyncSession;
 
+enum class SyncUserAdminMode {
+    // User is not an admin.
+    None,
+    // User wraps an admin token. Such users use the tokens they were configured
+    // with to directly open sessions, and do not make network requests.
+    WrapsAdminToken,
+    // User was marked by the authentication server as a user with admin privileges.
+    MarkedAsAdmin,
+};
```

Comment:
The terminology in this enum feels awkward. Maybe using an enum like `enum ObjectServerAccessLevel { Normal, Admin }` and having `SyncUser` store an `Optional<AccessLevel>` would be clearer? The `none` case would represent the user not being authenticated via the object server (i.e., having used an admin access token).

Or maybe it should just be `enum AccessLevel { Normal, Admin }`, and admin tokens / admin users should both be `Admin`, and the authenticated-with-object-server-vs-admin-token distinction tracked separately?